From 5bc7f1da3e3114873594e3d9138d7a6a03664788 Mon Sep 17 00:00:00 2001
From: Filip Filmar <filmil@gmail.com>
Date: Tue, 9 Dec 2025 07:37:05 +0000
Subject: [PATCH] Allow specifying significant dirs via env variables

The following can be set using environment variables:

- NVC_LIBEXEC_DIR
- NVC_LIB_DIR
- NVC_DATA_DIR

Useful for running NVC in sandboxed environments where the dir locations
are not predictable.
---
 src/util.c | 15 +++++++++++++++
 1 file changed, 15 insertions(+)

diff --git a/src/util.c b/src/util.c
index b72c85181..84f27aeac 100644
--- a/src/util.c
+++ b/src/util.c
@@ -2146,6 +2146,11 @@ static void get_relative_prefix(text_buf_t *tb)
 
 void get_libexec_dir(text_buf_t *tb)
 {
+   const char *env_dir = getenv("NVC_LIBEXEC_DIR");
+   if (env_dir != NULL) {
+      tb_cat(tb, env_dir);
+      return;
+   }
 #if defined __MINGW32__
    get_relative_prefix(tb);
    tb_cat(tb, DIR_SEP "libexec" DIR_SEP "nvc");
@@ -2156,6 +2161,11 @@ void get_libexec_dir(text_buf_t *tb)
 
 void get_lib_dir(text_buf_t *tb)
 {
+   const char *env_dir = getenv("NVC_LIB_DIR");
+   if (env_dir != NULL) {
+      tb_cat(tb, env_dir);
+      return;
+   }
 #if defined __MINGW32__
    get_relative_prefix(tb);
    tb_cat(tb, DIR_SEP "lib" DIR_SEP "nvc");
@@ -2166,6 +2176,11 @@ void get_lib_dir(text_buf_t *tb)
 
 void get_data_dir(text_buf_t *tb)
 {
+   const char *env_dir = getenv("NVC_DATA_DIR");
+   if (env_dir != NULL) {
+      tb_cat(tb, env_dir);
+      return;
+   }
 #if defined __MINGW32__
    get_relative_prefix(tb);
    tb_cat(tb, DIR_SEP "share" DIR_SEP "nvc");
-- 
2.34.1

